lua/barbar/ui/layout.lua

```diff
@@ -50,20 +49,20 @@ local SPACE_LEN = #' '
 --- @field width integer the amount of space allocated to the tabpage indicator
 
 --- @class barbar.ui.Layout
---- @field buffers integer[] different from `state.buffers` in that the `hide` option is respected. Only updated when calling `calculate_buffers_width`.
-local layout = { buffers = {} }
```

Comment:
The full state should be kept in `state`, makes things simple and debuggable.

Reply:
Sidenote but I don't like that our state is gaining fields but still relies on manual mutation for updating. Some sort of automated reactivity would probably be much less bug-prone. Don't have any solution for it though, and it might be overkill.

Reply:
We can make note of this to take a look at in the future. I agree that some of these fields would benefit from being self contained, and between the two of us I'm sure there's a way to make that happen :)